enhancements/dns/coredns-egressfirewall-integration.md

```diff
@@ -0,0 +1,761 @@
+corresponding to the wildcard DNS name. Otherwise, the plugin will update the `.status` with the regular DNS name and the corresponding current IP addresses
+along with the TTL and the next lookup time based on the TTL.
+* If the DNS lookup is for a wildcard DNS name and it matches with the `DNSNameResolver` CR corresponding to the wildcard DNS name,
+then the `ocp_dnsnameresolver` plugin will update the `.status` of the corresponding `DNSNameResolver` CR with the wildcard DNS name
+and the corresponding current IP addresses along with the TTL and the current lookup time as the last lookup time. If the wildcard DNS name's
+corresponding IP addresses and next lookup time (TTL + last lookup time) matches that of any other regular DNS name, added to the `.status`
+of the `DNSNameResolver` CR, then the details of the regular DNS name will be removed.
+* However, the updates will take place if there is a change in the next lookup time (TTL + last lookup time) for the DNS name or if the next
+lookup time is same for the DNS name, but the current IP addresses are different from the existing IP addresses. For the latter, the current
+IP addresses are added to the existing IP addresses of the DNS name. This will take care of the scenario where different CoreDNS pods get
+different subsets of the IP addresses in the response to the DNS lookup of the same DNS name. The next time to lookup will be same for them
+though the IP addresses may differ due to upstream DNS load balancing.
+* Additionally, the exact matching of the next lookup time may never be successful. If the existing next lookup time of a DNS name lies
+within a threshold value (say 5 seconds) of the current lookup time, specifically lies in the range defined by
+`[current lookup time - threshold duration, current lookup time + threshold duration]`, then they are considered as same.
+* The `Degraded` condition of the DNS name will be set to `false` along with the success reason and message.
+
+##### DNS lookup failure
+
+```shell
+update_dns_resolver_failure(objName, dnsName, response):
+	obj = get_dns_name_resolver(objName)
+	for each resolvedName in obj.status.resolvedNames:
+		if resolvedName.dnsName == dnsName:
+			if resolvedName.resolutionFailures == threshold:
+				remove(obj.status.resolvedNames, resolvedName)
+			else
+				resolvedName.ttl = min_ttl
+				resolvedName.lastLookupTime = now()
+				resolvedName.resolutionFailures = resolvedName.resolutionFailures + 1
+				cond = condition{type: "Degraded", status: true, reason: failure_reason(response), message: failure_message(response)}
+				add_condition(resolvedName.conditions, cond)
+	
+	update_status_dns_name_resolver(obj)
+```
+
+* If a DNS name (regular or wildcard) matches a `DNSNameResolver` CR, then the corresponding resolved name field in the `.status` will be updated.
+* If the corresponding `resolutionFailures` field has reached a threshold value (say 5), then the details of the DNS name will be removed from the
```

Comment:
Can we decide on a threshold as a group so that we can document it in the API? I think this is an important thing for a user to understand via the API documentation

Reply:
@Miciah @danwinship @npinaeva Can we take a call on the threshold value for `resolutionFailures`.

Reply:
I have set to to 5, both in the EP and the api godoc.